### DISCUSSION TITLE
What are all the game events to call services?

### DISCUSSION THREAD

**Discussion by tmob03** [ArgosOfIthica/scalu#23] [General]
Is there any way to detect the current map? Also I'm using this for GoldSrc if it really makes a difference (i don't think so?)

**Comment by tmob03** [comment]
```
sandbox my_sandbox

file {
	c1a0: c1a0
}

map {
	boot: @my_service
	c1a0: @c1a0
}

service my_service {
	[echo hello world!]
	[echo works in goldsrc]
}

service c1a0 {
	[echo the current map is c1a0!]
}
```

This is what I tried to load when the map was c1a0, but it didn't seem to work.

**Comment by ArgosOfIthaca** [comment]
Scalu was originally tested in TF2, Half-Life, and Quake, so it's very likely to be supported by all GoldSrc games. As for your syntax, you've created it correctly. The compiler will generate a file called c1a0.cfg which triggers the event you've specified. However, it is up to Source to actually execute the map specific config file, which not all games actually do. Team Fortress 2, for instance, does not actually allow map specific configuration files.

**Reply by tmob03** [reply]
I found game.cfg, which executes on map change, but also executes on save loads, hmm..

**Reply by chexo3** [reply]
Wait, TF2 doesn’t do map specific configs? I’ve literally done that before and it worked so I’m confused

**Reply by ArgosOfIthaca** [reply]
They were broken at the time, it's certainly possible they've been fixed. I don't closely follow TF2 development anymore.

**Reply by chexo3** [reply]
Fair.

**Comment by tmob03** [comment]
Actually, game.cfg works fine. I just need to detect when the player loads a save like this, and just subtract one from the current map counter when it detects it.
```
sandbox current_map

file {
	game: changedmap
}

bind {
	f1: loadedquick
}
map {
	boot: @init
	changedmap: @nextmap
	loadedquick: @loaded_save
}

service init {
	current_map = 0
}

service nextmap {
	current_map = current_map + 1
	if (current_map == 1){
		[echo current_map is 1]
		[say current map is 1]
	}
	if (current_map == 2){
		[echo current map is 2]
		[say current map is 2]
	}
	if (current_map == 3){
		[echo The curent map is 3]
		[say current_map is 3]
	}
	if (current_map == 4){
		[echo The curent map is 4]
		[say current map is 4]
	}
	if (current_map == 5){
		[echo current map is 5]
		[say current map is 5]
		[wait;wait;wait;wait;wait;wait;wait;wait;wait;wait;wait;disconnect]
	}
}

service loaded_save {
	if (current_map <= 0){
		current_map = current_map
}
else {
	current_map = current_map - 1
}
[load quick]
}
```
